filter/filter_impl/auth/consumer_sign.go

```diff
@@ -0,0 +1,39 @@
+package auth
+
+import (
+	"fmt"
+)
+import (
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/filter"
+	"github.com/apache/dubbo-go/protocol"
+)
+
+type ConsumerSignFilter struct {
+}
+
+func init() {
+	extension.SetFilter(constant.CONSUMER_SIGN_FILTER, getConsumerSignFilter)
+}
+
+func (filter *ConsumerSignFilter) Invoke(invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	logger.Infof("invoking ConsumerSign filter.")
+	url := invoker.GetUrl()
+	shouldAuth := url.GetParamBool(constant.SERVICE_AUTH_KEY, false)
```

Comment:
should u extract common method for auth? ProviderAuthFilter seems has the same logic

Reply:
It's hard to do that, I think.

Reply:
FIXED.